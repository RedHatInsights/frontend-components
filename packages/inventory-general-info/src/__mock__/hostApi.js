import { hosts } from '../api';
import MockAdapter from 'axios-mock-adapter';

export const mock = new MockAdapter(hosts.axios);
