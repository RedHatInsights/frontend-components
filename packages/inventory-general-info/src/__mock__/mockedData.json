{
    "total": 1,
    "count": 1,
    "page": 1,
    "per_page": 50,
    "results": [
        {
            "id": "6de30d78-f052-41d4-b123-125017443a98",
            "system_profile": {
                "arch": "x86_64",
                "cpu_flags": [ "fpu" ],
                "yum_repos": [
                    {
                        "name": "Extra Packages for Enterprise Linux 7 - Testing - $basearch",
                        "enabled": false,
                        "gpgcheck": true
                    }
                ],
                "os_release": "7.4",
                "bios_vendor": "SeaBIOS",
                "bios_version": "1.10.2-2.fc27",
                "kernel_modules": [ "veth" ],
                "last_boot_time": "2018-03-14T17:06:10",
                "number_of_cpus": 1,
                "cores_per_socket": 1,
                "enabled_services": [ "auditd" ],
                "bios_release_date": "04\/01\/2014",
                "number_of_sockets": 1,
                "os_kernel_version": "3.10.0",
                "running_processes": [ "qmgr" ],
                "satellite_managed": false,
                "installed_packages": [ "0:GeoIP-1.5.0-11.el7" ],
                "installed_services": [ "arp-ethers" ],
                "network_interfaces": [
                    {
                        "mtu": 65536,
                        "name": "lo",
                        "type": "loopback",
                        "state": "UNKNOWN",
                        "mac_address": "00:00:00:00:00:00",
                        "ipv4_addresses": [
                            "127.0.0.1"
                        ],
                        "ipv6_addresses": [
                            "::1"
                        ]
                    }
                ],
                "infrastructure_type": "virtual",
                "system_memory_bytes": 512073728,
                "infrastructure_vendor": "qemu"
            }
        }
    ]
}
