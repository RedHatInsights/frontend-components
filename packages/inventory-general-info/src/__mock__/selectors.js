/* eslint-disable camelcase */
export const testProperties = {
    number_of_cpus: 1,
    number_of_sockets: 1,
    cores_per_socket: 1,
    ramSize: '5 MB',
    disk_devices: []
};

export const osTest = {
    arch: 'test-arch',
    os_release: 'test-release',
    os_kernel_version: 'test-kernel',
    last_boot_time: 'test-boot',
    kernel_modules: []
};

export const biosTest = {
    bios_vendor: 'test-vendor',
    bios_version: 'test-version',
    bios_release_date: 1565702893431,
    cpu_flags: []
};

export const infraTest = {
    infrastructure_type: 'test-type',
    infrastructure_vendor: 'test-vendor',
    network: {
        ipv4: [ '1' ],
        ipv6: [ '6' ],
        interfaces: [ 'test' ]
    }
};

export const configTest = {
    installed_packages: [ 'packages' ],
    enabled_services: [ 'services' ],
    running_processes: [ 'processes' ],
    repositories: {
        enabled: [{
            base_url: 'test-url',
            name: 'test-name',
            enabled: true,
            gpgcheck: true
        }],
        disabled: []
    }
};

export const collectInfoTest = {
    insights_client_version: 'test-client',
    insights_egg_version: 'test-egg'
};
