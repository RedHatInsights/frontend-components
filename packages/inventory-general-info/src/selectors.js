/* eslint-disable camelcase */

function safeParser(toParse, key) {
    try {
        return JSON.parse(toParse);
    } catch {
        return { [key]: toParse };
    }
}

export const propertiesSelector = ({
    number_of_cpus,
    number_of_sockets,
    cores_per_socket,
    ramSize,
    disk_devices
} = {}) => ({
    cpuNumber: number_of_cpus,
    sockets: number_of_sockets,
    coresPerSocket: cores_per_socket,
    ramSize,
    storage: disk_devices && disk_devices.map(({ device, label, mount_point, options, type }) => ({
        ...device && safeParser(device, 'device'),
        label,
        ...mount_point && safeParser(mount_point, 'mountpoint'),
        ...options && safeParser(options, 'options'),
        ...type && safeParser(type, 'mounttype')
    })
    )
});

export const operatingSystem = ({
    arch,
    os_release,
    os_kernel_version,
    last_boot_time,
    kernel_modules
} = {}) => ({
    release: os_release,
    kernelRelease: os_kernel_version,
    architecture: arch,
    bootTime: last_boot_time,
    kernelModules: kernel_modules
});

export const biosSelector = ({
    bios_vendor,
    bios_version,
    bios_release_date,
    cpu_flags
} = {}) => ({
    vendor: bios_vendor,
    version: bios_version,
    releaseDate: bios_release_date && new Date(bios_release_date).toLocaleDateString(),
    csm: cpu_flags
});

export const infrastructureSelector = ({
    infrastructure_type,
    infrastructure_vendor,
    network = {}
} = {}) => ({
    type: infrastructure_type,
    vendor: infrastructure_vendor,
    ipv4: network.ipv4,
    ipv6: network.ipv6,
    nics: network.interfaces
});

export const configurationSelector = ({
    installed_packages,
    enabled_services,
    running_processes,
    repositories
} = {}) => ({
    packages: installed_packages,
    services: enabled_services,
    processes: running_processes,
    repositories
});

export const collectionInformationSelector = ({
    insights_client_version,
    insights_egg_version
} = {}) => ({
    client: insights_client_version,
    egg: insights_egg_version
});
