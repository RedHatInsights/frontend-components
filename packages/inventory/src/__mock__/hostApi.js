import { hosts, tags } from '../api';
import MockAdapter from 'axios-mock-adapter';

export const mock = new MockAdapter(hosts.axios);
export const mockTags = new MockAdapter(tags.axios);
