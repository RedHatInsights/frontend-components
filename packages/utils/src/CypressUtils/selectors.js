export const ouiaId = (id) => `[data-ouia-component-id="${id}"]`;

/** PF5 OUIA */
export const BUTTON = '[data-ouia-component-type="PF5/Button"]';
export const DROPDOWN = '[data-ouia-component-type="PF5/Dropdown"]';
export const DROPDOWN_TOGGLE = '[data-ouia-component-type="PF5/DropdownToggle"]';
export const DROPDOWN_ITEM = '[data-ouia-component-type="PF5/DropdownItem"]';
export const CHIP = '[data-ouia-component-type="PF5/Chip"]';
export const CHIP_GROUP = 'div[data-ouia-component-type="PF5/ChipGroup"]';
export const MENU = '[data-ouia-component-type="PF5/Menu"]';
export const MENU_TOGGLE_CHECKBOX = '[data-ouia-component-type="PF5/MenuToggleCheckbox"]';
export const MODAL_CONTENT = '[data-ouia-component-type="PF5/ModalContent"]';
export const CHECKBOX = '[data-ouia-component-type="PF5/Checkbox"]';
export const TITLE = '[data-ouia-component-type="PF5/Title"]';
export const PAGINATION = '[data-ouia-component-type="PF5/Pagination"]';
export const TEXT_INPUT = '[data-ouia-component-type="PF5/TextInput"]';
export const TOOLBAR = '[data-ouia-component-type="PF5/Toolbar"]';
export const TABLE = '[data-ouia-component-type="PF5/Table"]';
export const TABLE_HEADER = 'thead [data-ouia-component-type="PF5/TableRow"]';
export const TABLE_ROW = 'tbody [data-ouia-component-type="PF5/TableRow"]';
export const TABLE_ROW_CHECKBOX = 'tbody [data-ouia-component-type="PF5/TableRow"] input';
export const EMPTY_STATE = '[data-ouia-component-type="PF4/EmptyState"]';
export const CARD = '[data-ouia-component-type="PF5/Card"]';
export const BREADCRUMB = '[data-ouia-component-type="PF5/Breadcrumb"]';
export const TAB_CONTENT = '[data-ouia-component-type="PF5/TabContent"]';
export const TAB_BUTTON = '[data-ouia-component-type="PF5/TabButton"]';
export const ALERT = '[data-ouia-component-type="PF5/Alert"]';

/** PF5 classes */
export const MENU_TOGGLE = '.pf-v5-c-menu-toggle';
export const MENU_LIST = '.pf-v5-c-menu__list';
export const MENU_TOGGLE_TEXT = '.pf-v5-c-menu-toggle__text';
export const MENU_ITEM = '.pf-v5-c-menu__list-item';
export const SELECT_MENU_ITEM = '.pf-v5-c-select__menu-item';
export const PAGINATION_TOP = `${PAGINATION}:not(.pf-m-bottom)`;
export const PAGINATION_BOTTOM = `${PAGINATION}.pf-m-bottom`;
export const EMPTY_STATE_TITLE = '.pf-v5-c-empty-state__title';
export const EMPTY_STATE_ICON = '.pf-v5-c-empty-state__icon';
export const CARD_TITLE = '.pf-v5-c-card__title';

/** PF5 aria */
export const PAGINATION_NEXT = 'button[aria-label="Go to next page"]';
export const PAGINATION_PREV = 'button[aria-label="Go to previous page"]';
export const PAGINATION_LAST = 'button[aria-label="Go to last page"]';
export const PAGINATION_FIRST = 'button[aria-label="Go to first page"]';
export const PAGINATION_CURRENT = '[aria-label="Current page"]';

/** PrimaryToolbar OUIA */
export const PRIMARY_TOOLBAR = '[data-ouia-component-id="PrimaryToolbar"]';
export const PRIMARY_TOOLBAR_ACTIONS = '.ins-c-primary-toolbar__actions';
export const CONDITIONAL_FILTER = '[data-ouia-component-id="ConditionalFilter"]';
export const CONDITIONAL_FILTER_TOGGLE = '[data-ouia-component-id="ConditionalFilterToggle"]';
export const PT_CONDITIONAL_FILTER = '[data-ouia-component-id="ConditionalFilter"]';
export const PT_CONDITIONAL_FILTER_TOGGLE = '[data-ouia-component-id="ConditionalFilterToggle"]';
export const PT_CONDITIONAL_FILTER_LIST = '[data-ouia-component-id="ConditionalFilterList"]';
export const PT_BULK_SELECT = '[data-ouia-component-id="BulkSelect"]';
export const PT_BULK_SELECT_CHECKBOX = '[data-ouia-component-id="BulkSelectCheckbox"]';
export const PT_BULK_SELECT_LIST = '[data-ouia-component-id="BulkSelectList"]';
export const PT_BULK_ACTIONS_TOGGLE = '[data-ouia-component-id="BulkActionsToggle"]';
export const PT_BULK_ACTIONS_LIST = '[data-ouia-component-id="BulkActionsList"]';
