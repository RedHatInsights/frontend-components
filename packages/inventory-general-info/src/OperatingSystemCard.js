import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import LoadingCard from './LoadingCard';
import { generalMapper } from './dataMapper';
import { operatingSystem } from './selectors';
import { DateFormat } from '@redhat-cloud-services/frontend-components';
import { isDate } from './constants';

const OperatingSystemCard = ({ systemInfo, detailLoaded, handleClick }) => (
    <LoadingCard
        title="Operating system"
        isLoading={ !detailLoaded }
        items={ [
            { title: 'Release', value: systemInfo.release },
            { title: 'Kernel release', value: systemInfo.kernelRelease },
            { title: 'Architecture', value: systemInfo.architecture },
            { title: 'Last boot time', value: (isDate(systemInfo.bootTime) ?
                <DateFormat date={ systemInfo.bootTime } type="onlyDate" /> :
                'Not available'
            )
            },
            {
                title: 'Kernel modules',
                value: systemInfo.kernelModules ? `${systemInfo.kernelModules.length} modules` : 0,
                target: 'kernel_modules',
                onClick: () => {
                    handleClick(
                        'Kernel modules',
                        generalMapper(systemInfo.kernelModules, 'Module')
                    );
                }
            }
        ] }
    />
);

OperatingSystemCard.propTypes = {
    detailLoaded: PropTypes.bool,
    handleClick: PropTypes.func,
    systemInfo: PropTypes.shape({
        release: PropTypes.string,
        architecture: PropTypes.string,
        kernelRelease: PropTypes.string,
        bootTime: PropTypes.string,
        kernelModules: PropTypes.arrayOf(PropTypes.string)
    })
};
OperatingSystemCard.defaultProps = {
    detailLoaded: false,
    handleClick: () => undefined
};

export default connect(({
    entityDetails: {
        entity
    },
    systemProfileStore: {
        systemProfile
    }
}) => ({
    detailLoaded: systemProfile && systemProfile.loaded,
    systemInfo: operatingSystem(systemProfile, entity)
}))(OperatingSystemCard);
