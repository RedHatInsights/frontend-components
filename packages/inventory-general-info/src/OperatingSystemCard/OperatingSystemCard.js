import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import LoadingCard from '../LoadingCard';
import { generalMapper } from '../dataMapper';
import { operatingSystem } from '../selectors';
import DateFormat from '@redhat-cloud-services/frontend-components/DateFormat';
import { extraShape, isDate } from '../constants';

const OperatingSystemCard = ({
    systemInfo,
    detailLoaded,
    handleClick,
    hasRelease,
    hasKernelRelease,
    hasArchitecture,
    hasLastBoot,
    hasKernelModules,
    extra
}) => (
    <LoadingCard
        title="Operating system"
        isLoading={ !detailLoaded }
        items={ [
            ...hasRelease ? [{ title: 'Release', value: systemInfo.release }] : [],
            ...hasKernelRelease ? [{ title: 'Kernel release', value: systemInfo.kernelRelease }] : [],
            ...hasArchitecture ? [{ title: 'Architecture', value: systemInfo.architecture }] : [],
            ...hasLastBoot ? [{ title: 'Last boot time', value: (isDate(systemInfo.bootTime) ?
                <DateFormat date={ systemInfo.bootTime } type="onlyDate" /> :
                'Not available'
            )
            }] : [],
            ...hasKernelModules ? [{
                title: 'Kernel modules',
                value: systemInfo.kernelModules?.length,
                singular: 'module',
                target: 'kernel_modules',
                onClick: () => {
                    handleClick(
                        'Kernel modules',
                        generalMapper(systemInfo.kernelModules, 'Module')
                    );
                }
            }] : [],
            ...extra
        ] }
    />
);

OperatingSystemCard.propTypes = {
    detailLoaded: PropTypes.bool,
    handleClick: PropTypes.func,
    systemInfo: PropTypes.shape({
        release: PropTypes.string,
        architecture: PropTypes.string,
        kernelRelease: PropTypes.string,
        bootTime: PropTypes.string,
        kernelModules: PropTypes.arrayOf(PropTypes.string)
    }),
    hasRelease: PropTypes.bool,
    hasKernelRelease: PropTypes.bool,
    hasArchitecture: PropTypes.bool,
    hasLastBoot: PropTypes.bool,
    hasKernelModules: PropTypes.bool,
    extra: PropTypes.arrayOf(extraShape)
};
OperatingSystemCard.defaultProps = {
    detailLoaded: false,
    handleClick: () => undefined,
    hasRelease: true,
    hasKernelRelease: true,
    hasArchitecture: true,
    hasLastBoot: true,
    hasKernelModules: true,
    extra: []
};

export default connect(({
    entityDetails: {
        entity
    },
    systemProfileStore: {
        systemProfile
    }
}) => ({
    detailLoaded: systemProfile && systemProfile.loaded,
    systemInfo: operatingSystem(systemProfile, entity)
}))(OperatingSystemCard);
