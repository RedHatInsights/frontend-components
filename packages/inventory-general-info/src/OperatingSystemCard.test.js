/* eslint-disable camelcase */
import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import OperatingSystemCard from './OperatingSystemCard';
import configureStore from 'redux-mock-store';
import { osTest } from './__mock__/selectors';

describe('OperatingSystemCard', () => {
    let initialState;
    let mockStore;

    beforeEach(() => {
        mockStore = configureStore();
        initialState = {
            systemProfileStore: {
                systemProfile: {
                    loaded: true,
                    ...osTest
                }
            }
        };
    });

    it('should render correctly - no data', () => {
        const store = mockStore({ systemProfileStore: {}, entityDetails: {}});
        const wrapper = render(<OperatingSystemCard store={ store } />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render correctly with data', () => {
        const store = mockStore(initialState);
        const wrapper = render(<OperatingSystemCard store={ store } />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render enabled/disabled', () => {
        const store = mockStore({
            systemProfileStore: {
                systemProfile: {
                    loaded: true,
                    ...osTest
                }
            }
        });
        const wrapper = render(<OperatingSystemCard store={ store } />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    describe('api', () => {
        it('should NOT call handleClick', () => {
            const store = mockStore(initialState);
            const onClick = jest.fn();
            const wrapper = mount(<OperatingSystemCard store={ store } />);
            wrapper.find('TextListItem a').first().simulate('click');
            expect(onClick).not.toHaveBeenCalled();
        });

        it('should call handleClick on packages', () => {
            const store = mockStore(initialState);
            const onClick = jest.fn();
            const wrapper = mount(<OperatingSystemCard handleClick={ onClick } store={ store } />);
            wrapper.find('dd a').first().simulate('click');
            expect(onClick).toHaveBeenCalled();
        });
    });
});
