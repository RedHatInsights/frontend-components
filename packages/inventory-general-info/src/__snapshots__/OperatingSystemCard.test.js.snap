// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperatingSystemCard should render correctly - no data 1`] = `
<div
  class="pf-l-stack pf-m-gutter"
>
  <div
    class="pf-l-stack__item"
  >
    <div
      class="pf-c-content"
    >
      <h1
        class=""
        data-pf-content="true"
      >
        Operating System
      </h1>
    </div>
  </div>
  <div
    class="pf-l-stack__item pf-m-fill"
  >
    <div
      class="pf-c-content"
    >
      <dl
        class=""
        data-pf-content="true"
      >
        <dt
          class=""
          data-pf-content="true"
        >
          Release
        </dt>
        <dd
          class=""
          data-pf-content="true"
        >
          <div
            class="ins-c-skeleton ins-c-skeleton__sm"
          >
             
          </div>
        </dd>
        <dt
          class=""
          data-pf-content="true"
        >
          Kernel release
        </dt>
        <dd
          class=""
          data-pf-content="true"
        >
          <div
            class="ins-c-skeleton ins-c-skeleton__sm"
          >
             
          </div>
        </dd>
        <dt
          class=""
          data-pf-content="true"
        >
          Architecture
        </dt>
        <dd
          class=""
          data-pf-content="true"
        >
          <div
            class="ins-c-skeleton ins-c-skeleton__sm"
          >
             
          </div>
        </dd>
        <dt
          class=""
          data-pf-content="true"
        >
          Last boot time
        </dt>
        <dd
          class=""
          data-pf-content="true"
        >
          <div
            class="ins-c-skeleton ins-c-skeleton__sm"
          >
             
          </div>
        </dd>
        <dt
          class=""
          data-pf-content="true"
        >
          Kernel modules
        </dt>
        <dd
          class=""
          data-pf-content="true"
        >
          <div
            class="ins-c-skeleton ins-c-skeleton__sm"
          >
             
          </div>
        </dd>
      </dl>
    </div>
  </div>
</div>
`;

exports[`OperatingSystemCard should render correctly with data 1`] = `
<div
  class="pf-l-stack pf-m-gutter"
>
  <div
    class="pf-l-stack__item"
  >
    <div
      class="pf-c-content"
    >
      <h1
        class=""
        data-pf-content="true"
      >
        Operating System
      </h1>
    </div>
  </div>
  <div
    class="pf-l-stack__item pf-m-fill"
  >
    <div
      class="pf-c-content"
    >
      <dl
        class=""
        data-pf-content="true"
      >
        <dt
          class=""
          data-pf-content="true"
        >
          Release
        </dt>
        <dd
          class=""
          data-pf-content="true"
        >
          test-release
        </dd>
        <dt
          class=""
          data-pf-content="true"
        >
          Kernel release
        </dt>
        <dd
          class=""
          data-pf-content="true"
        >
          test-kernel
        </dd>
        <dt
          class=""
          data-pf-content="true"
        >
          Architecture
        </dt>
        <dd
          class=""
          data-pf-content="true"
        >
          test-arch
        </dd>
        <dt
          class=""
          data-pf-content="true"
        >
          Last boot time
        </dt>
        <dd
          class=""
          data-pf-content="true"
        >
          Invalid Date
        </dd>
        <dt
          class=""
          data-pf-content="true"
        >
          Kernel modules
        </dt>
        <dd
          class=""
          data-pf-content="true"
        >
          <a
            href="http://localhost:5000//kernel_modules"
          >
            0 modules
          </a>
        </dd>
      </dl>
    </div>
  </div>
</div>
`;

exports[`OperatingSystemCard should render enabled/disabled 1`] = `
<div
  class="pf-l-stack pf-m-gutter"
>
  <div
    class="pf-l-stack__item"
  >
    <div
      class="pf-c-content"
    >
      <h1
        class=""
        data-pf-content="true"
      >
        Operating System
      </h1>
    </div>
  </div>
  <div
    class="pf-l-stack__item pf-m-fill"
  >
    <div
      class="pf-c-content"
    >
      <dl
        class=""
        data-pf-content="true"
      >
        <dt
          class=""
          data-pf-content="true"
        >
          Release
        </dt>
        <dd
          class=""
          data-pf-content="true"
        >
          test-release
        </dd>
        <dt
          class=""
          data-pf-content="true"
        >
          Kernel release
        </dt>
        <dd
          class=""
          data-pf-content="true"
        >
          test-kernel
        </dd>
        <dt
          class=""
          data-pf-content="true"
        >
          Architecture
        </dt>
        <dd
          class=""
          data-pf-content="true"
        >
          test-arch
        </dd>
        <dt
          class=""
          data-pf-content="true"
        >
          Last boot time
        </dt>
        <dd
          class=""
          data-pf-content="true"
        >
          Invalid Date
        </dd>
        <dt
          class=""
          data-pf-content="true"
        >
          Kernel modules
        </dt>
        <dd
          class=""
          data-pf-content="true"
        >
          <a
            href="http://localhost:5000//kernel_modules"
          >
            0 modules
          </a>
        </dd>
      </dl>
    </div>
  </div>
</div>
`;
